design-documents/service-isolation/integration-tests.md

```diff
@@ -0,0 +1,46 @@
+# Integration Tests
+
+## Problem Statement
+
+Integration tests run in the same PHP process as Magento application.
+A test from one module may depend on a data fixture or a model from another module.
+In case of **single-area setup** or setup with **different application areas served by different instances**, integration tests and their dependencies may be deployed on **different instances** and, as a result, such tests won't be able to resolve their dependencies.
+
+The question: should integration testing framework support distributed deployment or keep it compatible to monolithic deployment only?
+
+Take into account future support of microservices architecture.
+
+## Purpose of Integration Tests in Magento
+
+Before deciding whether we should drop support of integration tests for distributed systems, let's revisit purpose of integration tests in Magento:
+1. validate extension points of a component (potential integration with other components)
+2. validate integration with other existing components. Integration tests don't use mocks (usually)
+3. validate integration with the environment. Integration tests run on installed application, with real DB, etc
```

Comment:
@maghamed : we should use integration tests for infra (e.g., service mesh) testing.
* @melnikovi : we probably need different type of tests

Reply:
https://github.com/magento/service-isolation-backlog-wiki/issues/32